client/src/Components/SelectionList.jsx

```diff
@@ -7,85 +7,72 @@ class SelectionList extends React.Component {
+    this.getSelectValue = this.getSelectValue.bind(this);
+    this.getOptionsValue = this.getOptionsValue.bind(this);
   }
 
   componentDidMount() {
-      console.log(this.getItemOptions(), 'this is what getItemOptions returns')
-    this.getItemOptions();
+    this.getSelectValue();
+    this.getOptionsValue();
   }
 
-  // getItemOptions(){
-  //   var selection = [];
-  //   getItemSelection(this.props.randomItemNumber, (err, data) => {
-  //     // if (err){
-  //     //   console.log(err, "err from selection list");
-  //     // return;
-  //     // } 
-  //     if (data[0]['options']){
-  //       var itemHasSelection = data[0]['options']
-  //       for (var keys in itemHasSelection){
-  //         selection.push(keys)
-  //       }
-  //       console.log(selection[0], 'before return statement')
-  //     }
-  //     return;
-    
-  //   })
-  //   console.log(selection[0], 'after return statement')  
-  //   return selection;
-  // }
 
-  getItemOptions(){
-    console.log(this.state.selectionPossibilties)
+  getSelectValue(){
     var selection = [];
     return getItemSelection(this.props.randomItemNumber)
       .then((data) => {
-        console.log(data, 'this is data')
-        console.log(data[0]['options'])
         if (data[0]['options']){
           var itemHasSelection = data[0]['options']
-          console.log(itemHasSelection, 'this is the data were iterating thorugh')
           for (var keys in itemHasSelection){
             selection.push(keys)
           }
         }
           console.log(this.state.selectionPossibilties, 'possibilities')
         console.log(selection, 'selection getting caught')
-        this.setState({
-          selectionPossibilties : selection
-        })
         // return selection;
+          this.setState({
+            selectionPossibilties : selection
+          })
       })
-      // .then((selection) => {
-      //   // return selection;
-      // })
-      // .then((selection) => {
-      //   selection.map(selections => {
-      //     console.log(selections, 'props are being passed down properly')
-      //   })
-      // })
-      .catch(() => {
-        console.log(selection, 'failed promise')
+      .catch((err) => {
+        console.log(err, 'failed promise')
       })
     }
-      // if (err){
-      //   console.log(err, "err from selection list");
-      // return;
-      // } 
-    
-
-
 
-//Need to send a getRequest to the server with the randomItemNumber, then map through the different selection values
-  //Afterwards can get the options with the same method
+  getOptionsValue(){
+    var options = [];
+    return getItemSelection(this.props.randomItemNumber)
+      .then((data) => {
+        console.log(data, 'data from options value')
+        if (data[0]['options']){
+          for (var key in data[0]['options']){
+            var individualOptions = [];
+            for (let i = 0; i < data[0]['options'][key].length; i++){
```

Comment:
Strongly recommend when you have large variable access names such as `data[0]['options'][key]` that you assign it to a easily readable variable name the line above where you need to use it.  For example, you could say `let option = data[0]['options'][key];` and then refer to it just as option in the `for` loop and in the `push` operator

it's definitely an art form for when to do this versus not, my general rule of thumb is if it's hard to read, I assign it a temporary variable so it's easy to determine whats going on

EDIT: Another alternative, you could have assigned a variable to `let varname = data[0]['options']` instead, and then just referred to that as you go down the chain, since you are rewriting that long term many times.  Later on you could say `varname[key]` for example.  Up to you on how you want to refactor this